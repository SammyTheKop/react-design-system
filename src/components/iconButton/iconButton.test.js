// react
import React, { useState } from "react";

// component
import IconButton from "./iconButton";

// jest renderer
import renderer from "react-test-renderer";

// testing library react
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

/**
 * @name TestComponent
 * @details use case of button component
 * @returns button component with the updated count
 */
const TestComponent = () => {
  const [count, setCounter] = useState(0);

  const handleButtonClick = () => {
    setCounter(count + 1);
  };

  return (
    <IconButton handleClick={handleButtonClick} type="image-icon">
      Click to increase: {count}
    </IconButton>
  );
};

// Case 1: IconButton Renders properly or not
it("Case 1: IconButton Component Renders Correctly", () => {
  const handleButtonClick = () => {
    console.log("IconButton Clicked");
  };

  const iconButtonComponentTree = renderer
    .create(<IconButton handleClick={handleButtonClick}>Test</IconButton>)
    .toJSON();
  expect(iconButtonComponentTree).toMatchSnapshot();
});

// Case 2: Checks the IconButton data-testid is correct or not
it("Case 2: Checks the IconButton data-testId is correct or not", () => {
  const handleButtonClick = () => {
    console.log("IconButton Clicked");
  };
  const { getByTestId } = render(
    <IconButton handleClick={handleButtonClick}>Test</IconButton>
  );
  expect(getByTestId("button")).toBeInTheDocument();
});

// Case 3: Checks the IconButton data-testid is correct or not
it("Case 3: Checks the IconButton Text data-testId is correct or not", () => {
  const handleButtonClick = () => {
    console.log("IconButton Clicked");
  };
  const { getByTestId } = render(
    <IconButton handleClick={handleButtonClick}>Test</IconButton>
  );
  expect(getByTestId("button-text")).toBeInTheDocument();
});

// Case 4: Checks the number of times button is clicked
it("Case 4: Checks the number of times button is clicked", () => {
  const { getByTestId, getByText } = render(<TestComponent />);
  fireEvent.click(getByTestId("button"));
  fireEvent.click(getByTestId("button"));
  expect(getByText("Click to increase: 2")).toBeInTheDocument();
});

// Case 5: Checks the IconButton Component Name is correct or Not
it("Case 5: Checks the IconButton Component Name is correct or Not", () => {
  const { getByText } = render(<TestComponent />);
  expect(getByText("Click to increase: 0")).toBeInTheDocument();
});
