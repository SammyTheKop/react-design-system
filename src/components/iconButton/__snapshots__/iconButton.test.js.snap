// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case 1: IconButton Component Renders Correctly 1`] = `
Array [
  <div
    className="button-component"
  >
    <button
      className=" icon-button-component button-medium button"
      data-testid="button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <p
        className="button-text"
        data-testid="button-text"
      >
        <div
          className="icon-button-component-contents"
        >
          <span
            className="icon-button-component-contents-icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </div>
      </p>
    </button>
  </div>,
  <style
    jsx={true}
  >
    
          /* Styles for the Button Component */
          .button-component .button {
            background: linear-gradient(#5aac44 0px, #519839 100%);
            box-shadow: 0 1px 0 #959da1;
            border-radius: 4px;
            border: 0px;
            color: white;
            height: 40px;
            padding: 8px;
            overflow: hidden;
          }

          .button-component .button:hover {
            background: #298fca;
            border-color: #298fca #298fca #026aa7;
          }

          .button-component .button-text {
            display: flex;
            justify-content: center;
            font-weight: 700;
            margin: 0;
          }

          .button-component .button-disabled {
            cursor: not-allowed;
            background: gray;
          }

          .button-component .button-disabled:hover {
            background-color: gray;
            border-color: gray;
          }

          /* Keep adding the Styles of the Button Size Props below */
          .button-component .button-extra-small {
            width: 50px;
          }

          .button-component .button-small {
            width: 100px;
          }

          .button-component .button-medium {
            width: 150px;
          }

          .button-component .button-large {
            width: 200px;
          }

          .button-component .button-extra-large {
            width: 250px;
          }

          .button-component .transparent {
            background: white;
            box-shadow: none;
            border-radius: none;
            color: black;
          }

          .button-component .button-red {
            color: #fff;
            background-color: #cb2431;
            background-image: linear-gradient(-180deg, #de4450, #cb2431 90%);
            border-color: rgba(27, 31, 35, 0.5);
          }

          .button-component .button-red:hover {
            color: #cb2431;
            background-color: #fafbfc;
            background-image: linear-gradient(-180deg, #fafbfc, #eff3f6 90%);
          }
        
  </style>,
  <style
    jsx={true}
  >
    
          /* Styles for the Icon Button Component */
          .icon-button-component-contents {
            display: flex;
            justify-content: space-around;
            width: 100%;
            align-items: center;
          }

          .icon-button-component-contents-text {
            font-weight: 700;
          }
        
  </style>,
]
`;
